bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/TeaserImpl.java

```diff
@@ -57,6 +83,49 @@ public String getLinkURL() {
         return super.getLinkURL();
     }
 
+    @PostConstruct
+    protected void initModel() {
+        super.initModel();
+        ValueMap properties = resource.getValueMap();
+        titleFromPage = properties.get(Teaser.PN_TITLE_FROM_PAGE, titleFromPage);
+        descriptionFromPage = properties.get(Teaser.PN_DESCRIPTION_FROM_PAGE, descriptionFromPage);
+    }
+
+    @Override
+    public String getTitle() {
+        if (this.title == null && !this.titleHidden) {
+            if (titleFromPage) {
+                this.title = this.getTargetPage()
+                        .map(tp -> StringUtils.defaultIfEmpty(tp.getPageTitle(), tp.getTitle()))
+                        .orElseGet(() -> this.getTeaserActions().stream().findFirst()
+                                .map(com.adobe.cq.wcm.core.components.internal.models.v1.TeaserImpl.Action::getTitle)
+                                .orElseGet(() -> Optional.ofNullable(getCurrentPage())
+                                        .map(cp -> StringUtils.defaultIfEmpty(cp.getPageTitle(), cp.getTitle()))
+                                        .orElse(null)));
+            } else {
+                this.title = this.resource.getValueMap().get(JcrConstants.JCR_TITLE, String.class);
+            }
+        }
+        return title;
+    }
+
+    @Override
+    public String getDescription() {
+        if (this.description == null && !this.descriptionHidden) {
+            if (descriptionFromPage) {
+                this.description = this.getTargetPage().map(Optional::of).orElseGet(() -> Optional.ofNullable(getCurrentPage()))
+                        .map(Page::getDescription)
+                        // page properties uses a plain text field - which may contain special chars that need to be escaped in HTML
+                        // because the resulting description from the teaser is expected to be HTML produced by the RTE editor
+                        .map(Text::escapeXml)
```

Comment:
Shouldn't we use https://sling.apache.org/apidocs/sling9/org/apache/sling/xss/XSSAPI.html#encodeForHTML-java.lang.String- instead?

Reply:
Doesn't Sightly take care of escaping the string in the HTML file? I wouldn't escape the string here as we don't know the context: it could be used in JSON and would need a different type of escaping.

Reply:
@jckautzmann , @bpauli mentioned that normally we expect the description to be edited with the RTE (which already does escape) so we skip HTL escaping. I too would love to be able to just rely on HTL, it's much safer.